packages/@aws-cdk/aws-appmesh/lib/virtual-gateway-listener.ts

```diff
@@ -76,11 +105,73 @@ export abstract class VirtualGatewayListener {
     return new GrpcGatewayListener(options);
   }
 
+  /**
+   * Protocol the listener implements
+   */
+  protected abstract protocol: Protocol;
+
+  /**
+   * Port to listen for connections on
+   */
+  protected abstract port: number;
+
+  /**
+   * Health checking strategy upstream nodes should use when communicating with the listener
+   */
+  protected abstract healthCheck?: HealthCheck;
+
+  /**
+   * The TLS mode.
+   *
+   * @default - none
+   */
+  readonly tlsMode?: TlsMode;
+
+  /**
+   * Represents the listener certificate
+   *
+   * @default - none
+   */
+  readonly tlsCertificate?: TlsCertificate;
+
   /**
    * Called when the GatewayListener type is initialized. Can be used to enforce
    * mutual exclusivity
    */
   public abstract bind(scope: cdk.Construct): VirtualGatewayListenerConfig;
+
+  protected renderHealthCheck(hc: HealthCheck): CfnVirtualGateway.VirtualGatewayHealthCheckPolicyProperty | undefined {
+    if (hc.protocol === Protocol.TCP) {
+      throw new Error('TCP health checks are not permitted for gateway listeners');
+    }
+
+    if (hc.protocol === Protocol.GRPC && hc.path) {
+      throw new Error('The path property cannot be set with Protocol.GRPC');
+    }
+
+    const protocol = hc.protocol? hc.protocol : this.protocol;
+
+    const healthCheck: CfnVirtualGateway.VirtualGatewayHealthCheckPolicyProperty = {
+      healthyThreshold: hc.healthyThreshold || 2,
+      intervalMillis: (hc.interval || cdk.Duration.seconds(5)).toMilliseconds(), // min
+      path: hc.path || ((protocol === Protocol.HTTP || protocol === Protocol.HTTP2) ? '/' : undefined),
+      port: hc.port || this.port,
+      protocol: hc.protocol || this.protocol,
+      timeoutMillis: (hc.timeout || cdk.Duration.seconds(2)).toMilliseconds(),
+      unhealthyThreshold: hc.unhealthyThreshold || 2,
+    };
+
+    validateHealthChecks(healthCheck);
+
+    return healthCheck;
+  }
+
+  protected renderTls(tlsCertificate: TlsCertificate): CfnVirtualGateway.VirtualGatewayListenerTlsProperty {
```

Comment:
Turn this into a function, instead of a `protected` method (I want to keep the surface area of this public class to a minimum).